x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -1590,6 +1594,25 @@ private static Expression castStringLiteral(Expression from, DataType target) {
                 return unresolvedAttribute(from, target.toString(), e);
             }
         }
+
+        private static Expression processVectorFunction(org.elasticsearch.xpack.esql.core.expression.function.Function vectorFunction) {
```

Comment:
Maybe I am wrong but it feels like there is a missing validation on the provided vector dimension (and maybe on the type of the vector components)?